### ISSUE TITLE
bazel mod tidy logs a warning for the very thing it's about to fix

### ISSUE THREAD

**Issue by Buzz-Lightyear** [bazelbuild/bazel#24495]
### Description of the bug:

bazel mod tidy logs a warning for the very thing it's about to fix, for e.g.:
```
> bazel mod tidy
WARNING: ...: The module extension go_deps defined in @gazelle//:extensions.bzl reported incorrect imports of repositories via use_repo():

Not imported, but reported as direct dependencies by the extension (may cause the build to fail):
    com_github_hashicorp_go_cleanhttp, com_linkedin_golang_c2s_backend_api_c2s_backend_go_api, com_linkedin_golang_gurney_gurney, com_linkedin_golang_rubiks_api_rubiks_go_api, in_golnkd_nimbus_inventory_manager

Fix the use_repo calls by running 'bazel mod tidy'.
INFO: Updated use_repo calls for @gazelle//:extensions.bzl%go_deps
```

Re-running doesn't log that warning because `bazel mod tidy` has already fixed it.

### Which category does this issue belong to?

Core

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run `bazel mod tidy` in a situation where you expect updates to `MODULE.bazel` (say while using any module extension that pulls in new dependencies).

### Which operating system are you running Bazel on?

Darwin

### What is the output of `bazel info release`?

release 7.1.2

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?

```text
Why is this relevant?
```

### Have you found anything relevant by searching the web?

No I haven't

### Any other information, logs, or outputs that you want to share?

Issue filed in Gazelle as well: https://github.com/bazel-contrib/bazel-gazelle/issues/1988

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.0.1 RC1](https://github.com/bazelbuild/bazel/releases/tag/8.0.1rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.0.1rc1. Thanks!

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.5.0 RC2](https://github.com/bazelbuild/bazel/releases/tag/7.5.0rc2). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.5.0rc2. Thanks!
